lib/utils/note-utils.ts

```diff
@@ -54,23 +54,26 @@ const getPreview = (content: string, searchQuery?: string) => {
 
     // use only the first term of a multi-term query
     const firstTerm = terms[0].toLocaleLowerCase();
-
-    const searchRegex = new RegExp(escapeRegExp(firstTerm), 'ims');
-    if (searchRegex.test(content)) {
-      const regExp = new RegExp(
-        '.{0,10}' + escapeRegExp(firstTerm) + '.{0,200}',
-        'ims'
-      );
-      const matches = regExp.exec(content);
-      if (matches && matches.length > 0) {
-        preview = matches[0];
-        preview = preview.replace(/ +/g, ' ').replace(/\n/g, '').trim();
-
-        // @todo: add '...' before/after to indicate truncation
-        return preview;
-      }
+    const leadingChars = 30 - firstTerm.length;
+
+    const regExp = new RegExp(
+      /* eslint-disable */
+      '(?<=\\s|^)[^\n]' + // split at a word boundary (pattern must be preceded by whitespace or beginning of string)
+      '{0,' +
+      leadingChars +
+      '}' + // up to leadingChars of text before the match
+        escapeRegExp(firstTerm) +
+        '.{0,200}(?=\\s|$)', // up to 200 characters of text after the match, splitting at a word boundary
+      'ims'
+      /* eslint-enable */
+    );
+    const matches = regExp.exec(content);
+    if (matches && matches.length > 0) {
+      preview = matches[0];
+      return preview;
```

Comment:
a quick fix for split surrogate pairs would be to extend to slice the first character if it's a low-surrogate, meaning it was removed from its high-surrogate. in these cases we'll get garbage displayed or the Unicode replacement character �

```ts
const isLowSurrogate = s => {
	const c = s.charCodeAt(0);

	return 0xDC00 <= c && c <= 0xDFFF;
}

…
return isLowSurrogate(preview)
	? preview.slice(1)
	: preview;
```